mailstore-connector/src/com/axonivy/connector/mailstore/MailSessionProvider.java

```diff
@@ -0,0 +1,99 @@
+package com.axonivy.connector.mailstore;
+
+import static com.axonivy.connector.mailstore.MailStoreService.LOG;
+
+import java.util.Properties;
+
+import javax.mail.Session;
+
+import org.apache.commons.lang3.BooleanUtils;
+
+import com.axonivy.connector.mailstore.enums.StartTLS;
+
+import ch.ivyteam.ivy.environment.Ivy;
+import ch.ivyteam.ivy.ssl.restricted.IvySslSocketFactory;
+import ch.ivyteam.ivy.ssl.restricted.SslClientSettings;
+import ch.ivyteam.ivy.ssl.restricted.SslConfig;
+import ch.ivyteam.ivy.vars.Variable;
+
+public class MailSessionProvider {
+
+	private static final String PROPERTIES_VAR = "properties";
+
+	private interface Property {
+		String MAIL_SMTP_SSL_SOCKET_FACTORY = "mail.smtp.ssl.socketFactory";
+		String MAIL_SMTP_SSL_SOCKET_FACTORY_FALLBACK = "mail.smtp.ssl.socketFactory.fallback";
+	}
+
+	static Session getSession(String storeName) throws Exception {
+		Properties properties = getProperties(storeName);
+		if (isStartTLSEnabled(properties)) {
+			properties.put(Property.MAIL_SMTP_SSL_SOCKET_FACTORY, ivySslContext());
+			// ensures that socket will only be created with our socket factory otherwise it
+			// will fail
+			properties.setProperty(Property.MAIL_SMTP_SSL_SOCKET_FACTORY_FALLBACK, "false");
+		}
+
+		return Session.getInstance(properties, null);
+	}
+
+	private static boolean isStartTLSEnabled(Properties properties) {
+		if (properties == null) {
+			return false;
+		}
+		return BooleanUtils.toBoolean(properties.getProperty(StartTLS.ENABLE.getProperty()))
+				|| BooleanUtils.toBoolean(properties.getProperty(StartTLS.REQUIRED.getProperty()));
+	}
+
+	private static IvySslSocketFactory ivySslContext() {
+		return new IvySslSocketFactory(new SslConfig(false, "noAlias", SslClientSettings.instance()));
+	}
```

Comment:
restricted internal code; but better than re-inventing it. We should get rid of it however, as soon as we have an official Mail-Client Api.